### ISSUE TITLE
Unexpected behaviour of SEQ and SNE

### ISSUE THREAD

**Issue by abu-co** [insou22/mipsy#284]
Consider the instruction
```assembly
seq	$t0, $t0, 42
```
which gets expanded into
```assembly
0x00400000 [0x2001002a]    addi   $at, $zero, 42
0x00400004 [0x34080001]    ori    $t0, $zero, 1
0x00400008 [0x11010002]    beq    $t0, $at, 2
0x0040000c [0x34080000]    ori    $t0, $zero, 0
```
due to this rule:
https://github.com/insou22/mipsy/blob/90ea695d02c69978a293182bd941d1d08a5b540b/mips.yaml#L1741-L1750

However, `$t0` will always end up being `1` as `1 != 42`, irrespective of the actual value it originally contains.

I wonder if this behaviour is correct?

**Comment by Dylan-Brotherston** [comment]
discussion here: #283

**Comment by Dylan-Brotherston** [comment]
Fixed in #287
